src/Tes/Repository/PostgreSqlCachingRepository.cs

```diff
@@ -97,6 +101,48 @@ protected async Task<IEnumerable<T>> GetItemsAsync(DbSet<T> dbSet, Expression<Fu
             return await _asyncPolicy.ExecuteAsync(ct => query.ToListAsync(ct), cancellationToken);
         }
 
+        protected async Task<IEnumerable<TesTaskDatabaseItem>> GetTesTaskDatabaseItemsByTagAsync(
+            TesDbContext dbContext,
+            Dictionary<string, string> tags,
+            CancellationToken cancellationToken)
+        {
+            ArgumentNullException.ThrowIfNull(dbContext);
+            ArgumentNullException.ThrowIfNull(tags);
+            if (tags.Count == 0) throw new ArgumentOutOfRangeException("Must specify more than one tag");
+
+            var sqlBuilder = new StringBuilder();
+            sqlBuilder.AppendLine("SELECT * FROM TesTasks WHERE ");
+            var tagConditions = tags.Select(kvp => $"\"Json\"->'Tags'->>'{kvp.Key}' = @p_{kvp.Key}");
+            sqlBuilder.AppendLine(string.Join(" AND ", tagConditions));
+
+            using var connection = dbContext.Database.GetDbConnection();
+            await connection.OpenAsync();
+
+            using var command = connection.CreateCommand();
+            command.CommandText = sqlBuilder.ToString();
+
+            foreach (var tag in tags)
+            {
+                command.Parameters.Add(new NpgsqlParameter($"p_{tag.Key}", tag.Value));
```

Comment:
According to the spec, if the provided tag.Value is empty, then the query should return all tasks with the provided tag.Key no matter what the value might be. This query does not appear to me to do that.